ledger-core/virtual/execute/execute.go

```diff
@@ -678,7 +681,13 @@ func (s *SMExecute) stepSendOutgoing(ctx smachine.ExecutionContext) smachine.Sta
 			return ctx.Error(throw.E("failed to publish bargeInCallback"))
 		}
 	} else {
-		s.outgoing.CallRequestFlags = payload.BuildCallRequestFlags(payload.SendResultDefault, payload.RepeatedCall)
+		if s.retryOutgoingCounter < MaxOutgoingRetries {
+			s.retryOutgoingCounter++
+			s.outgoing.CallRequestFlags = payload.BuildCallRequestFlags(payload.SendResultDefault, payload.RepeatedCall)
+		} else {
+			ctx.Log().Warn("outgoing retry attempts exceeded, stop processing SMExecute")
```

Comment:
```suggestion
			return ctx.Error("outgoing retry attempts exceeded, stop processing SMExecute")
```
Maybe with error will be better?